fiftyone/core/collections.py

```diff
@@ -11384,6 +11384,60 @@ def _get_label_attributes_schema(self, label_field):
 
         return schema
 
+    def _get_sidebar_group(self, group_name):
+        app_config = self._root_dataset.app_config
+        if app_config.sidebar_groups is None:
+            return None
+
+        for group in app_config.sidebar_groups:
+            if group.name == group_name:
+                return group
+
+        return None
+
+    def _has_sidebar_group(self, group_name):
+        return self._get_sidebar_group(group_name) is not None
+
+    def _add_paths_to_sidebar_group(self, paths, group_name, after_group=None):
+        dataset = self._root_dataset
+        dataset.app_config._add_paths_to_sidebar_group(
+            paths,
+            group_name,
+            after_group=after_group,
+            dataset=dataset,
+        )
+        dataset.save()
+
+    def _rename_sidebar_group(self, group_name, new_group_name):
+        dataset = self._root_dataset
+        if dataset.app_config.sidebar_groups is None:
+            return
+
+        existing_group = None
+        for group in dataset.app_config.sidebar_groups:
+            if group.name == new_group_name:
+                existing_group = group
+
+        for group in dataset.app_config.sidebar_groups[:]:
```

Comment:
is this doing anything?
```suggestion
        for group in dataset.app_config.sidebar_groups:
```

Reply:
Yeah it's subtle. `[:]` creates a copy of the list object in-memory so that I can `remove()` things from the original list while iterating over the copy in the loop.

Reply:
might be more clear to use `sidebar_groups.copy()` which would also create shallow copy like `[:]`

Reply:
Deal, but in return I get to complain when people use the walrus operator, which I think is unreadable 😄

Reply:
Okay, `feature_request := updated`